### ISSUE TITLE
get dataguide function needed by client

### ISSUE THREAD

**Issue by mbjones** [NCEAS/metacat#27]
---

Author Name: **Dan Higgins** (Dan Higgins)
Original Redmine Issue: 30, https://projects.ecoinformatics.org/ecoinfo/issues/30
Original Date: 2000-06-21
Original Assignee: Jivka Bojilova

---

The client needs to be able to ask the server for a LORE style dataguide for
each of the doctypes handled by the server (in order to show a treestructure
to guide structured query creation)

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Matt Jones** (Matt Jones)
Original Date: 2000-06-23T19:38:59Z

---

lowering priority because this is a search feature that requires the user to
understand complex queries and something about the semantics of the metadata
standards.  Will do, but not until more basic infrastructure is completed.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Jivka Bojilova** (Jivka Bojilova)
Original Date: 2000-07-27T16:02:59Z

---

assigned to Jivka Bojilova

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Jivka Bojilova** (Jivka Bojilova)
Original Date: 2000-07-28T21:46:59Z

---

We going to add doctype and docid attr to xml_index table first in order to be
able to retrieve the paths for a given doctype(s), if needed.
Currentlly implemented to retrieve all distinct, absolute paths in xml_index
table, ordered in the way they appear in the xml docs.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Jivka Bojilova** (Jivka Bojilova)
Original Date: 2000-08-09T01:02:59Z

---

DONE
Implemented as part of DBUtil class.
doctype and docid attr are already added to xml_index table.
Retrieves all distinct and absolute paths started with '/' for a given doctype
(if specified) ordered in the way they appear in the xml docs.
Are there need to retrive paths for a list of doctypes?

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Jivka Bojilova** (Jivka Bojilova)
Original Date: 2000-09-25T17:28:41Z

---

It is already DONE and used by dmanclient appl.
